x-pack/plugins/index_lifecycle_management/public/application/services/policies/policy_validation.ts

```diff
@@ -148,6 +158,7 @@ export const validatePolicy = (
   const hotPhaseErrors = validateHotPhase(policy.phases.hot);
   const warmPhaseErrors = validateWarmPhase(policy.phases.warm);
   const coldPhaseErrors = validateColdPhase(policy.phases.cold);
+  const frozenPhaseErrors = validateColdPhase(policy.phases.frozen);
```

Comment:
I think `validateFrozenPhase` function should be used here, but then again, it's weird why type_check didn't show any errors.

Reply:
TS uses structural type checking(https://www.typescriptlang.org/docs/handbook/type-compatibility.html). Structurally frozen and cold are the same at the moment which is why TS is not complaining. But you are right this should be aligned with their own functions/types as this is copy/pasta mistake.